cypress/e2e/edit-profile.cy.js

```diff
@@ -0,0 +1,49 @@
+describe('Edits profile details', () => {
+  it('should check that profile details are successfully edited', () => {
+    // Intercept the PUT request and login API call
+    cy.intercept('PUT', 'https://cors.noroff.dev/https://agency-api.noroff.dev/users/*').as(
+      'editProfile'
+    );
+    cy.intercept('POST', 'https://cors.noroff.dev/https://agency-api.noroff.dev/users/login').as(
+      'loginAPI'
+    );
+
+    cy.visit('http://127.0.0.1:4173/'); // Please replace this with your local vite url
+
+    // Login process
+    cy.contains('Log in').click();
+    cy.get('#email').type('thisuser@stud.noroff.no', { force: true });
+    cy.get('#password').type('qwe123***', { force: true });
+    cy.get('button[data-auth="login"]').click({ force: true });
+
+    // Navigate to profile edit modal
+    cy.contains('Profile').click();
+    cy.contains('EDIT PROFILE').click();
+
+    // Fill out the edit profile form
+    cy.wait(200);
```

Comment:
using .wait is in general not recommended, since we don't know how long the server will take to respond. A better pattern is to assert something that only exists on the page you're expecting. When doing so, cypress will look for the specific element until it timeouts, which is usually a better approach.

Reply:
Could i use `.should("be.visible")` for this? Without .wait the test wont properly type in the firstname field

Reply:
ah - i remember i had this issue with the bootstrap modal as well before! Didn't find a better solution for it in the end, so let's stick with the wait() unless you got any better ideas :)

Reply:
I suspect it is because of the very short animation for when the modal pops up. I'll stick to`.wait()` at least for now :p